distribution/pull_v2.go

```diff
@@ -147,6 +149,10 @@ func (ld *v2LayerDescriptor) DiffID() (layer.DiffID, error) {
 	return ld.V2MetadataService.GetDiffID(ld.digest)
 }
 
+func (ld *v2LayerDescriptor) ForeignSource() *distribution.Descriptor {
+	return ld.foreignSrc
```

Comment:
At this point, why does it matter if it is a foreign source? Shouldn't we just have a URL?